lib/flutter_opendroneid.dart

```diff
@@ -96,18 +105,28 @@ class FlutterOpenDroneId {
     });
     if (usedTechnologies == UsedTechnologies.Bluetooth ||
         usedTechnologies == UsedTechnologies.Both) {
-      await _api.startScanBluetooth();
+      final permissionGranted = await _ensureBluetoothPermissions();
+      if (permissionGranted) {
+        await _api.startScanBluetooth();
+      } else {
+        return Future.error('Permissions not granted');
```

Comment:
Pouzil bych klasicky `throw`, kdyz je to uvnitr `async` metody.

```suggestion
        throw Exception('Permissions not granted');
```